hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveSyncTool.java

```diff
@@ -194,11 +197,17 @@ private void syncSchema(String tableName, boolean tableExists, boolean useRealTi
       String outputFormatClassName = HoodieInputFormatUtils.getOutputFormatClassName(baseFileFormat);
       String serDeFormatClassName = HoodieInputFormatUtils.getSerDeClassName(baseFileFormat);
 
+      Map<String, String> serdeProperties = ConfigUtils.toMap(cfg.serdeProperties);
+      if (readAsOptimized) { // read optimized
+        serdeProperties.put(DefaultHoodieConfig.QUERY_TYPE_OPT_KEY, DefaultHoodieConfig.QUERY_TYPE_READ_OPTIMIZED_OPT_VAL);
+      } else { // read snapshot
+        serdeProperties.put(DefaultHoodieConfig.QUERY_TYPE_OPT_KEY, DefaultHoodieConfig.QUERY_TYPE_SNAPSHOT_OPT_VAL);
```

Comment:
Is there a difference in storing this in serde properties vs table properties ?

Reply:
Yes, the properties stored in the `serde` will convert to spark `options`  by spark datasource table.